lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -751,7 +756,22 @@ sub cluster_jobs {
         }
         else {
             push(@{$jobs->{$self->id}->{parallel_parents}}, $p->id);
-            $p->cluster_jobs($jobs);
+            my $cancelwhole = 1;
+            # check if the setting to disable cancelwhole is set
+            my $cwset = $p->settings_hash->{PARALLEL_CANCEL_WHOLE_CLUSTER};
+            $cancelwhole = 0 if (defined $cwset && $cwset eq '0');
+            if ($args{cancelmode} & !$cancelwhole) {
```

Comment:
Binary `&` intended here?

Reply:
yes to both of you, well spotted, thanks.